omegaconf/listconfig.py

```diff
@@ -66,6 +66,14 @@ def __init__(
         except Exception as ex:
             format_and_raise(node=None, key=None, value=None, cause=ex, msg=str(ex))
 
+    def _resolve_generic_ref_type(
+        self, ref_type: Optional[Type[Any]], element_type: Optional[Type[Any]]
+    ) -> Optional[Type[Any]]:
+        if ref_type == List:
+            return List[element_type]  # type: ignore
```

Comment:
Can you add a test that verifies that pickle works for Structured Configs?
I remember issues around serializing things like `List[int]`.

Reply:
There should be some tests already but I am not sure they are covering this properly.

Reply:
it looks like it doesn't work with picke, output: `Can't pickle typing.List[typing.Any]: it's not the same object as typing.List`

Reply:
In python 3.7 or above it works perfectly fine, in the other hand python 3.6 does not.

Reply:
I changed the code so it doesn't allow things like List[None] that I missed.

Reply:
I am still not seeing a test that verifies this.

Reply:
Let me re-cap one more time:
I asked about a test for pickling Structured Configs with annotations like List[int] and Dict[str, int].
The reason is that I remember some issues.

You said something like:
There was an issue with Python 3.6 but I worked around it.
I asked for the same test again, to be sure that your work around works.

This is not specifically about this PR, but as we are changing the internal representation of DictConfig and ListConfig we need to be sure that we don't introduce regressions.

Reply:
I'm going to push the test + the workaround for python3.6 ( I had to change the implementation of __getstate__ on basecontainer. I don't know if that's the best way to do it.

Reply:
let's chat.